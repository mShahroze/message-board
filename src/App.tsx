import React, { FC } from 'react';
import './app.scss';
import CommentsContainer from './components/comment/CommentsContainer';

const App: FC = () => {
  return (
    <div className='main'>
      <header>
        <h1>Welcome to the Message Board!</h1>
      </header>

      <section>
        <h1>Top Comments</h1>
        <CommentsContainer />
      </section>
    </div>
  );
};

export default App;
