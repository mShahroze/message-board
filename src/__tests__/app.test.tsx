/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { act, render } from '@testing-library/react';
import App from '../App';
import fetchMock from 'jest-fetch-mock';
import commentsMockData from '../testData/commentsData.json';

describe('App Component', () => {
  beforeEach(() => {
    fetchMock.mockResponseOnce(JSON.stringify(commentsMockData));
  });
  it('should take a snapshot', async () => {
    let component: any = { asFragment: null };
    await act(async () => {
      component = render(<App />);
    });
    const { asFragment } = component;
    expect(asFragment()).toMatchSnapshot();
  });
});
