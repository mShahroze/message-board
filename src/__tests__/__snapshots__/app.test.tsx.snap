// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <header>
      <h1>
        Welcome to the Message Board!
      </h1>
    </header>
    <section>
      <h1>
        Comments
      </h1>
      <div>
        <div
          class="comment"
        >
          <p>
            Could someone point out to me, what the benefit of winning the group is? I mean, both finalists of the group will qualify for the Spring Finals? So what is the point of fighting for 1st? Bragginâ€™ rights?
          </p>
          <span>
            MakeNBreak69
          </span>
          <div>
            View 2 Replies
          </div>
        </div>
        <div
          class="comment"
        >
          <p>
            NiP have the stream running on a big TV in the same room they're playing in.Are we doing this whole shenanigans again now? Didn't ESIC say there are gonna be consequences from now on? Getting a little tired of this.
          </p>
          <span>
            z0ttel89
          </span>
          <div>
            View 2 Replies
          </div>
        </div>
        <div
          class="comment"
        >
          <p>
            Blast has the best presentation and production of all the ongoing series at the moment, my favourite without a doubt. That sounded like a paid advertisement, but I really just like the series
          </p>
          <span>
            Spartan_exr
          </span>
          0
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
